controllers/azurecluster_controller.go

```diff
@@ -207,9 +207,11 @@ func (acr *AzureClusterReconciler) reconcileNormal(ctx context.Context, clusterS
 	log.Info("Reconciling AzureCluster")
 	azureCluster := clusterScope.AzureCluster
 
-	// If the AzureCluster doesn't have our finalizer, add it.
-	if controllerutil.AddFinalizer(azureCluster, infrav1.ClusterFinalizer) {
-		// Register the finalizer immediately to avoid orphaning Azure resources on delete
+	// Register our finalizer immediately to avoid orphaning Azure resources on delete
+	needsPatch := controllerutil.AddFinalizer(azureCluster, infrav1.ClusterFinalizer)
+	// Register the block-move annotation immediately to avoid moving un-paused ASO resources
+	needsPatch = AddBlockMoveAnnotation(azureCluster) || needsPatch
```

Comment:
Just to clarify, this just means we don't want to allow clusterctl move to happen before we're done reconciling, right?

Reply:
Right, this makes sure there isn't any interval of time where both an ASO resource has `reconcile-policy: manage` and the AzureCluster doesn't have the block-move label, which is the danger zone.